test/EFCore.OData.FunctionalTests/Query/ODataQueryTestFixtureInitializer.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.OData.Batch;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.AspNet.OData.Routing.Conventions;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.EntityFrameworkCore.TestUtilities;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.EntityFrameworkCore.Query
+{
+    public class ODataQueryTestFixtureInitializer
+    {
+        public static (string BaseAddress, IHttpClientFactory ClientFactory, IHost SelfHostServer) Initialize<TContext>(
+            string storeName,
+            Type[] controllers,
+            IEdmModel edmModel)
+            where TContext : DbContext
+        {
+            var port = PortArranger.Reserve();
+            var baseAddress = string.Format("http://localhost:{0}", port.ToString());
```

Comment:
Since you're not binding to IPv6 then you should directly address IPv4 to avoid connection delays.
```suggestion
            var baseAddress = string.Format("http://127.0.0.1:{0}", port.ToString());
```